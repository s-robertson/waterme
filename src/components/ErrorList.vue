<template>
  <div class="c-error-list" aria-live="assertive">
    <ul v-if="errors.length">
      <li v-for="(error, index) in errors" :key="index">{{ error }}</li>
    </ul>
  </div>
</template>

<script>
export default {
  name: "ErrorList",
  props: {
    errors: {
      type: Array,
      default: () => []
    }
  }
};
</script>
