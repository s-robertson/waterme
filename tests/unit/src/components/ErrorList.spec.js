import { createLocalVue, shallowMount } from "@vue/test-utils";
import ErrorList from "@/components/ErrorList";

const localVue = createLocalVue();

describe("ErrorList component", () => {
  it("Should render a list of error message", () => {
    const wrapper = shallowMount(ErrorList, {
      localVue,
      propsData: {
        errors: ["Error message one", "Error message two"]
      }
    });

    expect(wrapper.html()).toMatchSnapshot();
  });

  it("Should only render an aria-live region when there are no errors", () => {
    const wrapper = shallowMount(ErrorList, {
      localVue
    });

    expect(wrapper.html()).toMatchSnapshot();
  });
});
