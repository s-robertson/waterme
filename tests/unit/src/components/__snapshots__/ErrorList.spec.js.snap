// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorList component Should only render an aria-live region when there are no errors 1`] = `
<div aria-live="assertive" class="c-error-list">
  <!---->
</div>
`;

exports[`ErrorList component Should render a list of error message 1`] = `
<div aria-live="assertive" class="c-error-list">
  <ul>
    <li>Error message one</li>
    <li>Error message two</li>
  </ul>
</div>
`;
